clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -163,14 +166,18 @@ protected DelegateConsumerRecordResult delegateConsumerRecord(
   protected void putInStorageEngine(int partition, byte[] keyBytes, Put put) {
     try {
       // TODO: Honor BatchConflictResolutionPolicy and maybe persist RMD for batch push records.
-      switch (getStorageOperationType(partition, put.putValue, put.replicationMetadataPayload)) {
+      StorageOperationType storageOperationType =
+          getStorageOperationType(partition, put.putValue, put.replicationMetadataPayload);
+      switch (storageOperationType) {
         case VALUE_AND_RMD:
```

Comment:
The above strategy won't work for follower replica..
Another solution will be delaying this optimization in the persistency layer.
So when the ingestion task is trying to persist a chunking manifest (which can be inferred via the schema id), it needs to deserialize it to check how many chunks inside, and if there is only one, the ingestion task needs to perform the optimization:
1. Persist a non-chunked value.
2. Remove previously persisted chunks.

We don't have to do this at the very beginning, but it is good to have a metric to measure how many such kind of entries will have per store and this can be done in `LeaderProducerCallback`.

Reply:
I think I can understand the intention and thanks to Lincong there is already a ticket for this optimization. We can definitely work on this to optimize the perf. 
But I am wondering whether adding more temporary metric is going to be helpful here if we are determined to do this in the near future...and I am hesitated because we already have so many metrics in the ingestion path. If we are determined to do this, I'd suggest just do it in the next step when this whole implementation is done.

Reply:
If we will do this as the next step, the metric won't be necessary.